File: TypeScript/Problem #2 [Hard]/main.ts
let input: Array<number> = [1, 2, 3, 4, 5]

let product_of_list = (list: Array<number>, index: number): number => {
    let product: number = 1

    for(let n of list) {
        if (list.indexOf(n) != index)
            product *= n
    }

    return product
}

let solve = (input: Array<number>): Array<number> => {
    let output: Array<number> = []

    for(let n of input) {
        let product: number = product_of_list(input, input.indexOf(n))
        output.push(product)
    }

    return output;
}
        
solve(input)

File: Python/Problem #2 [Hard]/main.py
_input = [3, 4, -1, 1]

def product_of_list(n_list, curr_index):
    product = 1

    for i in range(len(n_list)):
        if i != curr_index:
            product *= n_list[i]
    
    return product

def solve(_input):
    output = []

    for i in range(len(_input)):
        product = product_of_list(_input, i)
        output.append(product)
    
    return output
        
print(solve(_input))